src/Kafker/Helpers/EmitPreserveTime.cs

```diff
@@ -0,0 +1,133 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Data;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Xml;
+using Kafker.Commands;
+using Kafker.Configurations;
+using Kafker.Kafka;
+using McMaster.Extensions.CommandLineUtils;
+
+namespace Kafker.Helpers
+{
+    public class EmitPreserveTime : Emit
+    {
+        static volatile bool startEvent = false;
+        private readonly IConsole _console;
+        private static IProducerFactory _producerFactory;
+        private readonly KafkerSettings _settings;
+
+        public EmitPreserveTime(IConsole console, IProducerFactory producerFactory,
+            KafkerSettings settings) : base(console, producerFactory, settings)
+        {
+            _console = console;
+            _producerFactory = producerFactory;
+            _settings = settings;
+        }
+
+        public override async Task<int> EmitEvents(CancellationToken cancellationToken, string fileName, string topic)
+        {
+            var cfg = await ExtractorHelper.ReadConfigurationAsync(topic, _settings, _console);
+            var producedEvents = 0;
+
+            try
+            {
+                var getDictionary = await CreatePreserveTime(fileName);
```

Comment:
Break this method `CreatePreserveTime` to more with dedicated responsibilities: 

1. load from file all data into list of tuples <timestamp, json>
2. create time map for events, which receives this list and returns dictionary

Name the dictionary somehow more relevant. For example, `eventsMappedByTime`